.github/workflows/release.yml

```diff
@@ -204,36 +216,40 @@ jobs:
 
         - name: Build archives
+          if: matrix.os != 'windows-latest'
           shell: bash
           run: |
             IFS=',' read -ra BIN_NAMES <<< "${{ env.BIN_NAMES }}"
             ASSETS=""
             for BIN_NAME in "${BIN_NAMES[@]}"; do
               staging="$BIN_NAME-${{ needs.create-release.outputs.release_version }}-${{ matrix.cargo_targets }}"
               mkdir -p "$staging"
-              if [ "${{ matrix.os }}" = "windows-latest" ]; then
-                cp "target/${{ matrix.cargo_targets }}/optimized-release/$BIN_NAME.exe" "$staging/"
-                cd "$staging"
-                7z a "../$staging.zip" .
-                ASSETS+="$staging.zip,"
-              else
-                cp "target/${{ matrix.cargo_targets }}/optimized-release/$BIN_NAME" "$staging/"
-                tar czf "$staging.tar.gz" -C "$staging" .
-                ASSETS+="$staging.tar.gz,"
-              fi
+              cp "target/${{ matrix.cargo_targets }}/optimized-release/$BIN_NAME" "$staging/"
+              tar czf "$staging.tar.gz" -C "$staging" .
+              ASSETS+="$staging.tar.gz,"
             done
-            echo "ASSET=${ASSETS::-1}" >> $GITHUB_ENV
+            echo "ASSET=$(echo $ASSETS | sed 's/,$//')" >> $GITHUB_ENV
 
-        - name: Upload release archives
-          if: github.event.inputs.upload_artifacts == 'true' || github.event_name == 'push'
+        - name: Build archives (windows)
+          if: matrix.os == 'windows-latest'
+          shell: pwsh
           run: |
-            IFS=',' read -ra ASSETS <<< "${{ env.ASSET }}"
-            for ASSET in "${ASSETS[@]}"; do
-              ASSET_NAME=$(basename $ASSET)
-              curl \
-                -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
-                -H "Content-Type: $(file -b --mime-type application/octet-stream)" \
-                --data-binary @"$ASSET" \
-                "${{ needs.create-release.outputs.upload_url }}?name=$ASSET_NAME"
-            done
-        
\ No newline at end of file
+            $BIN_NAMES = "${{ env.BIN_NAMES }}".Split(',')
+            $ASSETS = @()
+            foreach ($BIN_NAME in $BIN_NAMES) {
+              $staging = "$BIN_NAME-${{ needs.create-release.outputs.release_version }}-${{ matrix.cargo_targets }}"
+              New-Item -ItemType Directory -Force -Path "$staging"
+              Copy-Item -Path "target/${{ matrix.cargo_targets }}/optimized-release/$BIN_NAME.exe" -Destination "$staging/"
+              Set-Location -Path "$staging"
+              Compress-Archive -Path * -DestinationPath "../$staging.zip"
+              $ASSETS += "$staging.zip"
+              Set-Location -Path ..
+            }
+            $ASSETS = $ASSETS -join ','
+            Add-Content -Path $env:GITHUB_ENV -Value "ASSET=$ASSETS"
+
+        - uses: n0-computer/actions-upload-release-asset@main
```

Comment:
I'm in general a bit careful about using actions from a branch rather than a tag, which can still be moving like .e.g. `v1`.  But the repo doesn't seem to have any tags.